homeassistant/components/verisure/climate.py

```diff
@@ -0,0 +1,218 @@
+"""Support for Verisure heatpump."""
+import logging
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_FAN_MODE, SUPPORT_OPERATION_MODE,
+    SUPPORT_SWING_MODE, SUPPORT_ON_OFF)
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+from homeassistant.components.verisure import HUB as hub
+from homeassistant.components.verisure import CONF_CLIMATE
+
+_LOGGER = logging.getLogger(__name__)
+HEAT_PUMPS = None
+VERISIRE_HASS_OP_MODE = {
+    'AUTO': 'auto',
+    'FAN': 'fan_only',
+    'COOL': 'cool',
+    'DRY': 'dry',
+    'HEAT': 'heat'
+}
+
+HASS_VERISURE_OP_MODE = {
+    'auto': 'AUTO',
+    'fan_only': 'FAN',
+    'cool': 'COOL',
+    'dry': 'DRY',
+    'heat': 'HEAT'
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Verisure heatpump."""
+    import jsonpath
+    jsonpath = jsonpath.jsonpath
+    global HEAT_PUMPS
+    hub.update_overview()
+    if int(hub.config.get(CONF_CLIMATE, 1)):
+        HEAT_PUMPS = hub.get('$.heatPumps')
+        if HEAT_PUMPS:
+            for heat_pump in HEAT_PUMPS[0]:
+                device_label = jsonpath(heat_pump, '$.deviceLabel')[0]
+                add_entities([
```

Comment:
Collect created entities in a list and call `add_entities` only once with all the entities.